projects/sandbox/datagen/datagen/scripts/glitches.py

```diff
@@ -4,121 +4,240 @@
-            # restrict to triggers within current data chunk
-            data = data.resample(sample_rate)
-            data = data[channel]
-            times = data.times.value
-            mask = (triggers["time"] > times[0] + window) & (
-                triggers["time"] < times[-1] - window
+    if len(triggers) == 0:
+        return
+
+    # set 'start' and 'stop' so we aren't querying unnecessary data
+    start = np.min(triggers["time"]) - pad[0] - 1
+    stop = np.max(triggers["time"]) + pad[1] - 1
+
+    logging.info(
+        f"Querying {len(triggers)} triggers from {trigger_path}"
+        f" over {stop - start} seconds of data "
+    )
+
+    data = TimeSeries.get(
+        channel,
+        start=start,
+        end=stop,
+    )
+    data = data.resample(sample_rate)
+
+    # reset start / stop based on what was actually found
+    start, stop = data.times[0].value, data.times[-1].value
+
+    # query data for each trigger
+    for trigger in triggers:
+        time = trigger["time"]
+        beg, end = time - pad[0], time + pad[1]
+
+        if end > stop or beg < start:
+            logging.warning(
+                f"Trigger at time {time} is too close to end of segment."
+                f" Skipping"
             )
-            chunk_triggers = triggers[mask]
-            # query data for each trigger
-            for trigger in chunk_triggers:
-                time = trigger["time"]
-                try:
-                    glitch_ts = data.crop(time - window, time + window)
-                except ValueError:
-                    logging.warning(
-                        f"Data not available for trigger at time: {time}"
-                    )
-                    continue
-                else:
-                    glitches.append(list(glitch_ts.value))
-                    snrs.append(trigger["snr"])
-                    gpstimes.append(time)
-
-    glitches = np.stack(glitches)
-    return glitches, snrs, gpstimes
+            continue
+
+        try:
+            glitch_ts = data.crop(beg, end)
+        except ValueError:
+            logging.warning(f"Data not available for trigger at time: {time}")
+            continue
+        else:
+            print(len(glitch_ts.value))
+            glitches.append(list(glitch_ts.value))
+            snrs.append(trigger["snr"])
+            gpstimes.append(time)
+
+    if glitches:
+        glitches = np.stack(glitches)
+        with h5py.File(outfile, "w") as f:
+            f.create_dataset("glitches", data=glitches)
+            f.create_dataset("snrs", data=snrs)
+            f.create_dataset("times", data=gpstimes)
+
+    return outfile
+
+
+def deploy_collect_glitches(
+    trigger_paths: List[Path],
+    pad: Tuple[float, float],
+    snr_thresh: float,
+    sample_rate: float,
+    channel: str,
+    ifo: str,
+    outdir: Path,
+    condordir: Path,
+    accounting_group: str,
+    accounting_group_user: str,
+    request_memory: float = 32678,
```

Comment:
Should be 32768 for a power of 2, not that it actually makes a difference

Reply:
Good eye